Lesson 3/Flask View + API call.py

```diff
@@ -0,0 +1,174 @@
+
+    url = "https://test.bitpay.com/currencies"
+
+    headers = {
+        "accept": "application/json",
+        "Content-Type": "application/json",
+        "X-Accept-Version": "2.0.0"
+    }
+    symbol = ''
+
+    response = requests.get(url, headers=headers).json()
+    for item in response['data']:
+        if item['code'] == currency:
+            symbol = item['symbol']
+    symbol_btc = response['data'][0]['symbol']
+
+    return (f'{convert} {symbol_btc} <---->  BTC<br>'
+            f'{converter} {symbol} <---->  {currency}')
+
+
+@app.route("/order-price")
+@use_kwargs(
+    {
+        'country': fields.Str(
+            load_default='BillingCountry'
+        )
+    },
+    location='query'
+)
+def order_price(country: str):
+    query = (f'SELECT BillingCountry, sum(UnitPrice) '
+             f'FROM invoices JOIN invoice_items ON invoices.InvoiceId = invoice_items.InvoiceId '
+             f'GROUP BY BillingCountry HAVING BillingCountry = "{country}"')
+    result = execute_query(query=query)
+    return format_records(result)
+
+
+@app.route("/get-all-info-about-track")
+def get_all_info_about_track():
+    query = (f'SELECT albums."AlbumId", albums."Title", SUM(tracks.Milliseconds / 1000 / 3600.0) as SUM '
+             f'FROM playlist_track '
+             f'JOIN playlists ON playlist_track.PlaylistId = playlists.PlaylistId '
+             f'JOIN tracks ON playlist_track.TrackId = tracks.TrackId '
+             f'JOIN media_types ON media_types.MediaTypeId = tracks.MediaTypeId '
+             f'JOIN genres ON genres.GenreId = tracks.GenreId '
+             f'JOIN albums ON albums.AlbumId = tracks.AlbumId '
+             f'JOIN artists ON artists.ArtistId = albums.ArtistId '
+             f'JOIN invoice_items ON tracks.TrackId = invoice_items.TrackId'
+             f' JOIN invoices ON invoice_items.InvoiceId = invoices.InvoiceId '
+             f'JOIN customers ON customers.CustomerId = invoices.CustomerId '
+             f'JOIN employees ON employees.EmployeeId = customers.SupportRepId '
+             f'GROUP BY albums.AlbumId, albums.Title;')
+    result = execute_query(query=query)
+    return transform_time(result)
+
+@app.route("/stats-by-city")
+@use_kwargs(
+    {
+        'ganre': fields.Str(
+            load_default='Jazz'
+        )
+    },
+    location='query'
+)
+def stats_by_city(ganre: str):
+    query = (f'SELECT '
+             f'genres."Name" as Ganre, invoices."BillingCity" as City, '
+             f'invoices."BillingCountry" as Country, count(genres."Name") as listen '
+             f'FROM tracks '
+             f'JOIN genres ON tracks."GenreId" = genres."GenreId" '
+             f'JOIN invoice_items ON tracks."TrackId" = invoice_items."InvoiceId" '
+             f'JOIN invoices ON invoice_items."InvoiceId" = invoices."InvoiceId" '
+             f'GROUP BY genres."Name", invoices."BillingCity", invoices."BillingCountry" HAVING genres."Name" = "{ganre}" '
```

Comment:
there is a case when two cities will have the same value, please fix it based on this info